plonky2/src/plonk/circuit_data.rs

```diff
@@ -234,6 +234,8 @@ pub struct CommonCircuitData<
 
     pub(crate) num_virtual_targets: usize,
 
+    pub(crate) num_public_inputs: usize,
```

Comment:
I think we could add a check
```rust
assert_eq!(proof_with_pis.public_inputs.len(), common_data.num_public_inputs);
```
in `crate::verifer::verify` and `CircuitBuilder::verify_proof_with_pis`, now that this is part of the common data.

Reply:
Done